src/oauth2-client/controller/collection.ts

```diff
@@ -17,11 +20,65 @@ class ClientCollectionController extends Controller {
       }
     }
 
-    const clients = await findClientsByUser(user);
+    const clients = await findByUser(user);
     ctx.response.body = hal.collection(user, clients);
 
   }
 
+  async post(ctx: Context) {
+
+    const user = await userService.findById(ctx.state.params.id);
+    if (!await privilegeService.hasPrivilege(ctx, 'admin')) {
+      throw new Forbidden('Only users with the "admin" privilege can inspect OAuth2 clients that are not your own');
+    }
+    const allowedGrantTypes = ctx.request.body.allowedGrantTypes;
+    let clientId = ctx.request.body.clientId;
+
+    const redirectUris = ctx.request.body.redirectUris.split(' ');
+
+    if (!clientId) {
+      clientId = randomId(10);
+    } else if (clientId.length < 6) {
+      throw new UnprocessableEntity('clientId must be at least 6 characters or left empty');
+    }
+
+    const clientSecret = randomId(20);
```

Comment:
nit: Doesn't really matter, but it doesn't look like we need this variable until line 69.